### PR TITLE
Refactor MISP parameters handling and validation

### LINKED ISSUE #996: `parse_parameters()` does not match the connector parameter schema.
`parse_parameters()` does not match the connector parameter schema. The schema supplies `SSL_CHECK`, `PROXY_SERVER`, `ADDITIONAL_HEADERS`, `USER_AGENT`, and `USE_GLOBAL_PROXY`, while the worker reads `SSL`, `PROXIES`, and `HEADERS`. As a result, SSL verification and proxy/header settings are silently ignored. 

Please normalize these using the same parsing behavior as `MispCollector`, and parse `REQUEST_TIMEOUT` from its stored string with a safe default. Add a test using the real connector fixture, whose timeout is currently an empty string.

_edited scope_

_Originally posted by @armins in https://github.com/taranis-ai/taranis-ai/pull/984#pullrequestreview-4905437464_

### PR DESCRIPTION
**Pull request by not4Pedro**
- Introduced `parse_misp_runtime_parameters` function to streamline parameter parsing for MISP connectors and collectors.
- Updated `MispCollector` and `MispConnector` to utilize the new parameter parsing logic.
- Added validation for `REQUEST_TIMEOUT` to ensure it is a positive integer.
- Enhanced `MISPParameters` class to include validation rules for parameters.
- Updated tests to cover new parameter handling and validation scenarios.

### CODE REVIEW

File: src/models/models/admin.py [hunk 1]
```diff
@@ -177,6 +178,58 @@ class ParameterValue(TaranisBaseModel):
     parameter: str = ""
     value: str | None = ""
 
+    @classmethod
+    def validate_rules(cls, value: str | None, rules: list[str]) -> None:
```
[the remark this answers is not in the document]
Comment (b3n4kh) [reply src/models/models/admin.py]: I also don't really like the naming here and would suggest something even more direct.

```python
   @classmethod
   def _ensure_positive_int(value: str):
       pass # add actual code

   @classmethod
   def _ensure_json_object(value: str):
       pass # add actual code

    @field_validator("REQUEST_TIMEOUT")
    @classmethod
    def validate_request_timeout(cls, value: str) -> str:
        return cls._ensure_positive_int(value)

    @field_validator("ADDITIONAL_HEADERS")
    @classmethod
    def validate_additional_headers(cls, value: str) -> str:
        return cls._ensure_json_object(value)
```

File: src/worker/worker/misp_parameters.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import json
+from typing import Any
+
+
+def parse_misp_runtime_parameters(parameters: dict[str, Any]) -> tuple[bool, dict[str, str] | None, dict[str, Any], int]:
+    headers: dict[str, Any] = {}
+    if additional_headers := parameters.get("ADDITIONAL_HEADERS"):
```
[the remark this answers is not in the document]
Comment (not4Pedro) [reply src/worker/worker/misp_parameters.py:7]: I’d keep validation in src/models rather than repeat it in the worker helper. MISP payloads are normalised and validated by `MISPParameters` before they reach the worker, and the relevant rules are also defined in the preseed data. The worker helper should only convert those validated values for PyMISP.

File: src/core/tests/unit/test_misp_parameters.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+@pytest.mark.parametrize(
+    "normalize,worker_type",
+    [
+        (OSINTSource.get_with_defaults, "misp_collector"),
+        (Connector.get_with_defaults, "misp_connector"),
+    ],
+)
+def test_misp_worker_parameters_apply_defaults_and_global_proxy(monkeypatch, normalize, worker_type):
+    monkeypatch.setattr(
+        Settings,
```
[the remark this answers is not in the document]
Comment (not4Pedro) [reply src/core/tests/unit/test_misp_parameters.py:39]: This behaviour is already covered by `test_proxy_selection_matrix` in the worker API tests. That test includes non-MISP RSS sources with the global proxy both enabled and disabled. I don’t think another unit test would add meaningful coverage.

File: src/models/models/admin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (not4Pedro) [reply src/models/models/admin.py:182]: I’d keep the string rule names here. These rules also come from the preseed data, where they are stored as strings. Changing the model to use callables would give us two different ways to represent the same rules and make the flow harder to follow.

**Review by b3n4kh — CHANGES_REQUESTED** [review]
Obsoleted by #1008

**Comment by not4Pedro** [comment]
obsolete
